pkg/daemon/ceph/osd/volume.go

```diff
@@ -79,9 +80,26 @@ func (a *OsdAgent) initializeDevices(context *clusterd.Context, devices *DeviceO
 		strconv.Itoa(a.storeConfig.OSDsPerDevice),
 	}...)
 
-	// ceph-volume is soon implementing a parameter to specify the "fast devices", which correspond to the "metadataDevice" from the
-	// crd spec. After that is implemented, we can implement this. In the meantime, we fall back to use rook's partitioning.
+	if a.storeConfig.StoreType == config.Bluestore && a.storeConfig.DatabaseSizeMB > 0 {
+		if a.storeConfig.DatabaseSizeMB < 1024 {
+			// ceph-volume will convert this value to ?G. It needs to be > 1G to invoke lvcreate.
+			logger.Infof("skipping databaseSizeMB setting. For it should be larger than 1024MB.")
+		} else {
+			batchArgs = append(batchArgs, []string{
+				databaseSizeFlag,
+				// ceph-volume takes in this value in bytes
+				strconv.FormatInt(int64(a.storeConfig.DatabaseSizeMB)*1024*1024, 10),
+			}...)
+		}
+	}
+
+	// When mixed hdd/ssd devices are given, ceph-volume configures db lv on the ssd.
 	metadataDeviceSpecified := false
+	if a.metadataDevice != "" {
```

Comment:
Specifying the metadataDevice is added to ceph-volume in v14.1.0
https://github.com/ceph/ceph/commit/0a16dbd7a8f597cde6debbdcd968cdaf44f6ce47

Reply:
c-v changes are typically also backported to luminous and mimic. Any idea on the backport status? It would be good to at least document which versions of ceph can use this feature, otherwise specifying a metadataDevice would fail, right?

Reply:
apparently it is already backported as https://github.com/ceph/ceph/pull/24208

Reply:
For the metadataDevice & databaseSizeMB of ceph-volume backport status, I checked the code, it's backported to v12.2.10. Basically all versions supports `c-v lvm batch --prepare` supports medataDevice & databaseSizeMB and we have a check for that in `getCephVolumeSupported` methods. 

I didn't test Ceph v14 because they didn't release the docker images and the official rook supported versions are `v13.2.x` & `v12.2.x`

Reply:
As for the specifying the metadataDevice of c-v, it didn't backported in v12 & v13.